src/linker/Linker.Dataflow/FlowAnnotations.cs

```diff
@@ -0,0 +1,340 @@
+						continue;
+					}
+
+					annotatedFields.Add (new FieldAnnotation (field, annotation));
+				}
+			}
+
+			var annotatedMethods = new ArrayBuilder<MethodAnnotations> ();
+
+			// Next go over all methods with an explicit annotation
+			if (type.HasMethods) {
+				foreach (MethodDefinition method in type.Methods) {
+					DynamicallyAccessedMemberKinds [] paramAnnotations = null;
+
+					// We convert indices from metadata space to IL space here.
+					// IL space assigns index 0 to the `this` parameter on instance methods.
+					int offset = method.HasThis && !method.ExplicitThis ? 1 : 0;
+
+					for (int i = 0; i < method.Parameters.Count; i++) {
+						
+						if (!IsTypeInterestingForDataflow (method.Parameters [i].ParameterType)) {
+							continue;
+						}
+
+						DynamicallyAccessedMemberKinds pa = _source.GetParameterAnnotation (method, i);
+						if (pa == 0) {
+							continue;
+						}
+
+						if (paramAnnotations == null) {
+							paramAnnotations = new DynamicallyAccessedMemberKinds [method.Parameters.Count + offset];
+						}
+						paramAnnotations [i + offset] = pa;
+					}
+
+					// TODO: add special magic for instance methods on System.Type
+
+					DynamicallyAccessedMemberKinds returnAnnotation = IsTypeInterestingForDataflow(method.ReturnType) ?
+						_source.GetReturnParameterAnnotation (method) : 0;
+					if (returnAnnotation != 0 || paramAnnotations != null) {
+						annotatedMethods.Add (new MethodAnnotations (method, paramAnnotations, returnAnnotation));
+					}
+				}
+			}
+
+			// Next up are properties. Annotations on properties are kind of meta because we need to
+			// map them to annotations on methods/fields. They're syntactic sugar - what they do is expressible
+			// by placing attribute on the accessor/backing field. For complex properties, that's what people
+			// will need to do anyway. Like so:
+			//
+			// [field: Attribute]
+			// Type MyProperty {
+			//     [return: Attribute]
+			//     get;
+			//     [value: Attribute]
+			//     set;
+			//  }
+			//
+
+			if (type.HasProperties) {
+				foreach (PropertyDefinition property in type.Properties) {
+					
+					if (!IsTypeInterestingForDataflow (property.PropertyType)) {
+						continue;
+					}
+
+					DynamicallyAccessedMemberKinds annotation = _source.GetPropertyAnnotation (property);
+					if (annotation == 0) {
+						continue;
+					}
+
+					FieldDefinition backingFieldFromSetter = null;
+
+					// Propagate the annotation to the setter method
+					MethodDefinition setMethod = property.SetMethod;
+					if (setMethod != null) {
+
+						if (!ScanMethodBodyForFieldAccess (setMethod.Body, write: true, out backingFieldFromSetter)) {
+							// TODO: warn we couldn't find a unique backing field
+						}
+
+						if (annotatedMethods.Any (a => a.Method == setMethod)) {
+							// TODO: warn: duplicate annotation. not propagating.
+						} else {
+							int offset = setMethod.HasThis && !setMethod.ExplicitThis ? 1 : 0;
```

Comment:
Can you please create a helper for this - the code to propagate annotations will also need this - and it would be nice to have a "better" name for this operation.